### ISSUE TITLE
Crash TypeError: 'NoneType' object has no attribute '__getitem__'

### ISSUE THREAD

**Issue by masgo** [kdeldycke/maildir-deduplicate#24]
I am getting a reproducible error:

I use this command
python __init__.py -o /path-to-my-dir/getmaildir/

This is the whole output (yes, there are a lot of mails in that dir)

```bash
Processing 460581 mails in /mnt/rraid/COMPRESSED/Backup/Mails/getmaildir .........................Traceback (most recent call last):
  File "__init__.py", line 572, in <module>
    main()
  File "__init__.py", line 516, in main
    duplicates_run(opts, maildir_paths)
  File "__init__.py", line 534, in duplicates_run
    mail_count += collate_folder_by_hash(mails_by_hash, maildir, opts.message_id)
  File "__init__.py", line 287, in collate_folder_by_hash
    mail_hash, header_text = compute_hash_key(mail_file, message, use_message_id)
  File "__init__.py", line 271, in compute_hash_key
    canonical_headers_text = get_canonical_headers(mail_file, message)
  File "__init__.py", line 182, in get_canonical_headers
    canonical_value = get_canonical_header_value(header, value)
  File "__init__.py", line 241, in get_canonical_header_value
    utc_timestamp = email.utils.mktime_tz(parsed)
  File "/usr/lib/python2.7/email/_parseaddr.py", line 154, in mktime_tz
    if data[9] is None:
TypeError: 'NoneType' object has no attribute '__getitem__'
```
any ideas?

(update: the error message)

**Comment by aspiers** [comment]
Please can you add some debug to find out what the file / header / value in question are?

**Comment by masgo** [comment]
How can I add it? It is obviously very hard to find the file that caused the problems. Can I set some debug flags for the script? (I have no experience with python)

**Comment by aspiers** [comment]
Without Python experience, another thing you could do is narrow it down to a single mail by binary search:

1. split the mailbox in half
2. find out which half has the problem
3. using the half with the problem, go back to step 1 and repeat

If you then share the guilty mail with us, that would greatly accelerate a fix.

**Comment by masgo** [comment]
Is there another way? The inbox has 460581 mails in it and is NOT on an SSD (because it is a backup mailbox). Doing simple things like `du -sh` or `ls -1 | wc -l` takes minutes because of the low IOPS of the rotating disks.

**Comment by aspiers** [comment]
Yes, you could pay a Python developer :)  I am pretty sure it will be easy to fix.

**Comment by masgo** [comment]
I meant more like: output the current processed file .. then the one previous to the crash is causing it

**Comment by aspiers** [comment]
That's why my first suggestion was to add debug to the code to show which file is the problem.

**Comment by masgo** [comment]
I got the bad mail. Python is not so hard after all. Then I did binary search on the mail and found the Date header to be the bad boy. This minimal example crashes the script:
```
From: test@example.com
To: test@example.com
Subject: test
Date: 15/05/2013 15:59

test
```

**Comment by kdeldycke** [comment]
Fixed by #33.
